community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -227,10 +236,10 @@ public void run()
         {
             rotationExecutor.execute( runnable );
         }
-        catch ( RejectedExecutionException e )
+        catch ( Exception e )
         {
-            // Run directly instead
-            runnable.run();
+            rotationListener.rotationError( e, outRef.get() );
```

Comment:
about point 2, it's not a failure to rotate, it points to a bug in around scheduling a rotation, which is something else. Anyways, I think I'm done with the shed now, sporting dashing colors :)